### PR TITLE
Convert tests to TypeScript

### PR DESCRIPTION
**Pull request by fregante**
Pros:

- JSX is now handled by TypeScript in tests as well; this drops babel

Cons:

- the lines reported by AVA are incorrect because AVA is run on the bundle

Suggestions welcome about the position of the source tests folder

### CODE REVIEW

File: tests/_source/index.tsx [hunk 1]
```diff
@@ -159,16 +159,17 @@ test.serial('render svg', t => {
...
 	t.truthy(element);
-	t.true(document.createElementNS.calledTwice);
+	t.true(createElementNSSpy.calledTwice);
 
 	const xmlns = 'http://www.w3.org/2000/svg';
-	t.deepEqual(document.createElementNS.firstCall.args, [xmlns, 'text']);
-	t.deepEqual(document.createElementNS.secondCall.args, [xmlns, 'svg']);
+	t.deepEqual(createElementNSSpy.firstCall.args, [xmlns, 'text']);
+	t.deepEqual(createElementNSSpy.secondCall.args, [xmlns, 'svg']);
+	createElementNSSpy.restore();
```
Comment (fregante) [inline tests/_source/index.tsx:167]: Suggestions wanted to avoid having to manually create and restore each spy explicitly.

**Comment by vadimdemedes** [comment]
Why do tests need to be in a `_source` folder and have their own package.json?

**Comment by fregante** [comment]
The build exists due to: https://github.com/vadimdemedes/dom-chef/pull/68

> The problematic combination is:
> 
> * JSX in test files
> * `type: module` in main module
> 
> Babel is needed to support JSX in test files, but there's no way to then import `./index.js` since it's a `type: module` that can't be `require`'d


And I moved the tests to a subdirectory to avoid having to configure AVA to point to a new ugly `.built` folder. Let me see what the alternative is.

**Comment by fregante** [comment]
It turns out it didn't require any AVA config (tsx files are not normally included), it was just TypeScript that required some extra config to avoid:

```
[!] (plugin typescript) Error: @rollup/plugin-typescript: 'outDir' must be located inside 'dir'.
```

**Comment by fregante** [comment]
I'm specifically avoiding to also convert `_fixtures.ts` because it'd just have to contain a bunch of `as any` and `ts-ignore` since it's just simulating the DOM in a Node environment.

**Pull request merged**